src/Sulu/Bundle/AdminBundle/Resources/js/containers/ProfileFormOverlay/ProfileFormOverlay.js

```diff
@@ -25,21 +24,17 @@ const RESOURCE_KEY = 'profile';
 class ProfileFormOverlay extends React.Component<Props> {
     formRef: ?Form;
     title: string;
-    @observable formStore: MemoryFormStore;
+    @observable formStore: FormStoreInterface;
     saving: boolean = false;
 
     constructor(props: Props) {
         super(props);
 
-        Promise.all([
-            metadataStore.getSchema(FORM_KEY),
-            metadataStore.getJsonSchema(FORM_KEY),
-            ResourceRequester.get(RESOURCE_KEY),
-        ]).then(this.handleResponse);
+        ResourceRequester.get(RESOURCE_KEY).then(this.handleResponse);
     }
 
-    @action handleResponse = ([schema, jsonSchema, data]: [RawSchema, Object, Object]) => {
-        this.formStore = new MemoryFormStore(data, schema, jsonSchema);
+    @action handleResponse = (data: Object) => {
+        this.formStore = memoryFormStoreFactory.createFromFormKey(FORM_KEY, data);
```

Comment:
@nnatter @alexander-schranz I think this here is the interesting part. No need to handle the schema asynchronously on your own, and still an easy way to create the store, while still being able to passing it to the `Form` component.

Reply:
I am kind of curious why this is implemented using a `MemoryFormStore` when the data is loaded from the server anyway 😅 
But beside of that, the solution looks good to me 🙂

Reply:
@nnatter At first I thought the same, but then I remembered that this is loading the profile, which is not a standard REST interface, because it does not use an `id` parameter in the URL but data from the session to know which user to load.